src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -104,6 +105,15 @@ def create_container(cmd,
 
     container_resource_requirements = _create_resource_requirements(cpu=cpu, memory=memory)
 
+    if (len(image.split('/')) > 1) and image.split('/')[0].endswith('.azurecr.io'):
+
+     registry_login_server, registry_username, registry_password = get_login_credentials(
```

Comment:
Just to confirm, this method looks for the registry in the same subscription as the container instance.

Also if `registry_username` or `registry_password` are specified, this will override them. This might result in unexpected behavior when users want to access their registry using an SP. You probably can pass them and let `get_login_credentials` handle that for you.

Another concern is that the credential here will expire in one hour.